Sanitizer.cmake

```diff
@@ -0,0 +1,50 @@
+# Used to turn on certain sanitizers, if available:
+# 
+# clang:
+#    -dumpcompiler: 4.2.1
+#       address
+#       thread
+#       undefined
+# gcc:
+#    -dumpcompiler: 4.8.3
+#       address
+#       thread
+#
+# Note: This modifies CMAKE_CXX_FLAGS and CMAKE_EXE_LINKER_FLAGS
+# Usage: add this when doing cmake configuration: -DSANITIZER=address
+
+option(SANITIZER "Sanitizer: one of address/thread/undefined, depending on compiler" OFF)
```

Comment:
```Provide an option for the user to select as ON or OFF.```

Can't have the described values. What are you trying to achieve? Why not sanitizer on/off, if on add flags?